sway/layout.c

```diff
@@ -279,19 +283,53 @@ void move_container(swayc_t *container, enum movement_direction dir) {
 		sway_log(L_DEBUG, "container:%p, parent:%p, child %p,",
 				container,parent,child);
 		if (parent->layout == layout
+			|| layout == L_NONE /* accept any layout for next/prev direction */
```

Comment:
"layout == L_NONE" was to broad, replaced by "layout == L_NONE && parent->type == C_CONTAINER". This is intended to accept the MOVE_PREV/MOVE_NEXT directions regardless of the actual container layout.